src/Cake.Common/Build/BuildSystem.cs

```diff
@@ -136,7 +144,9 @@ public sealed class BuildSystem
                 | (GitLabCI.IsRunningOnGitLabCI ? BuildProvider.GitLabCI : BuildProvider.Local)
                 | (TFBuild.IsRunningOnAzurePipelines ? BuildProvider.AzurePipelines : BuildProvider.Local)
                 | (TFBuild.IsRunningOnAzurePipelinesHosted ? BuildProvider.AzurePipelinesHosted : BuildProvider.Local)
```

Comment:
I think we can remove `TFBuild` here since `AzurePipelines` already should return the same value.